grakn-engine/src/main/java/ai/grakn/engine/factory/EngineGraknGraphFactory.java

```diff
@@ -50,19 +54,29 @@ public static EngineGraknGraphFactory create(Properties properties) {
     private EngineGraknGraphFactory(Properties properties) {
         this.properties = new Properties();
         this.properties.putAll(properties);
+        this.engineURI = properties.getProperty(GraknEngineConfig.SERVER_HOST_NAME) + ":" + properties.getProperty(GraknEngineConfig.SERVER_PORT_NUMBER);
+        this.systemKeyspace = new SystemKeyspace(this);
     }
 
     public synchronized void refreshConnections(){
         FactoryBuilder.refresh();
     }
 
     public GraknGraph getGraph(String keyspace, GraknTxType type){
-        return FactoryBuilder.getFactory(keyspace, Grakn.DEFAULT_URI, properties).open(type);
+        if(!keyspace.equals(SystemKeyspace.SYSTEM_GRAPH_NAME)) {
```

Comment:
Just to point something out. This is what is going to guarantee the load order. So in the end wether we use a real singleton or a static class or whatever the load order has to be ensured somewhere.